src/audio/module_adapter/module/generic.c

```diff
@@ -336,6 +336,9 @@ int module_reset(struct processing_module *mod)
 	if (md->state < MODULE_IDLE)
 		return 0;
 #endif
+	/* cancel task if DP task*/
+	if (mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP && mod->dev->task)
```

Comment:
Is this check safe? I mean task is cancelled, where is task set to null? It's initially NULL, so this check will cover DP tasks never started, but how about a DP task that has already ended?

Reply:
If the task is cancel - its OK, calling cancel again is safe and won't have any effect
If task if freed: 
```
static inline void comp_free(struct comp_dev *dev)
{
	assert(dev->drv->ops.free);

	/* free task if shared component or DP task*/
	if ((dev->is_shared || dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP) &&
	    dev->task) {
		schedule_task_free(dev->task);
		rfree(dev->task);
		dev->task = NULL;
	}

	dev->drv->ops.free(dev);
}
```

free and reset are both called from IPC thread, so it looks safe. Anyway, I can change it to safer:
```
		task * _task = dev->task;
		dev->task = NULL;
		schedule_task_free(_task);
		rfree(_task);
```

Reply:
@marcinszkudlinski @lyakh I was actually thinking of the case where "dev->task" is a dangling pointer. I couldn't immediately find where dev->task is zeroed, so not sure if this check can ensure task is a valid task object.

Reply:
@lyakh you're right with IPC serialization, no need for double-over-protection.  @kv2019i  - as you see pointer is zeroized when task is freed, No changes required here